optuna/samplers/_tpe/sampler.py

```diff
@@ -233,6 +310,38 @@ def _split_observation_pairs(
         above = np.asarray([v for v in above if v is not None], dtype=float)
         return below, above
 
+    def _split_multivariate_observation_pairs(
+        self,
+        config_vals: Dict[str, List[Optional[float]]],
+        loss_vals: List[Tuple[float, float]],
+    ) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+
+        config_vals = {k: np.asarray(v, dtype=float) for k, v in config_vals.items()}
+        loss_vals = np.asarray(loss_vals, dtype=[("step", float), ("score", float)])
+
+        # TODO(kstoneriv3): Change of the order of 1. exclusion of `None` and 2. splitting might
+        # be needed. Independent sampler in `TPESampler` first splits the observations and
+        # then exclude `None`.
```

Comment:
I don't know why the order of "excluding None" and "splitting" is changed. Could you explain the reason for this? The current one leads to the small estimation of `n_below`.

Reply:
There are pros and cons to this implementation.

### Pros of exclusion of `None` -> splitting
- simpler code (exclusion for both `above` and `below` makes repeated code, so probably not good for maintenance)
- avoiding `below` to have zero samples (in the case all samples of `below` contains `None` we basically sample from prior)
### Cons of exclusion of `None` -> splitting
- lead to small estimation of n_below if the effective number of samples (samples that do not contain None) is small
- different from current implementations of univariate cases

I do not mean to insist on my implementation, but I would like to hear opinions on this to see which is more suitable.

Reply:
Thanks for the clarification. Your implementation is reasonable to me. By the way, currently, we only support the `IntersectionSearchSpace` as a built-in search space. So, this is not so big a matter. I think we can accept the current implementation.

Reply:
OK, thank you for your response. Then I will leave this as it is.